injection/sharedmain/main.go

```diff
@@ -313,6 +313,11 @@ func MainWithConfig(ctx context.Context, component string, cfg *rest.Config, cto
 		return controller.StartAll(ctx, controllers...)
 	})
 
+	// Setup default health checks to catch issues with cache sync etc.
+	if !healthProbesDisabled(ctx) {
+		injection.ServeHealthProbes(ctx)
```

Comment:
I think we should make this part of the `errgroup` so that if it fails to start it should shut everything down

Reply:
Ok makes sense we can also avoid some of the go routine nesting.